crates/indicators/src/average/wma.rs

```diff
@@ -142,21 +158,16 @@ impl MovingAverage for WeightedMovingAverage {
     fn count(&self) -> usize {
         self.inputs.len()
     }
+
     fn update_raw(&mut self, value: f64) {
-        if !self.has_inputs {
-            self.has_inputs = true;
-            self.inputs.push(value);
-            self.value = value;
-            return;
-        }
         if self.inputs.len() == self.period {
-            self.inputs.remove(0);
+            self.inputs.pop_front();
```

Comment:
If we're using a `VecDeque` with a fixed capacity then this might not be necessary?

Reply:
`VecDeque::with_capacity(n)` only **pre-allocates** space; it doesn’t enforce a hard cap.
If you keep pushing after `len == n`, the deque re-allocates and grows, so you must still `pop_front()` (or otherwise drop the oldest element) to keep the window at exactly `period` items.